fiftyone/utils/utils3d.py

```diff
@@ -0,0 +1,394 @@
+
+logger = logging.getLogger(__name__)
+
+Vec2Tuple = Tuple[int, int]
+Vec3Tuple = Tuple[int, int, int]
+
+shading_gradient_map = {
+    # lowest value is red
+    0.0: (165, 0, 38),
+    0.111: (215, 48, 39),
+    0.222: (244, 109, 67),
+    0.333: (253, 174, 97),
+    0.444: (254, 224, 144),
+    # mid-value is gray
+    0.555: (224, 243, 248),
+    0.666: (171, 217, 233),
+    0.777: (116, 173, 209),
+    0.888: (69, 117, 180),
+    # highest value is blue
+    1.0: (49, 54, 149),
+}
+
+
+class OrthographicProjectionMetadata(DynamicEmbeddedDocument):
+    """
+    Metadata class to store orthographic projection results
+    """
+
+    filepath = fof.StringField()
+    min_bound = fof.ListField(fof.FloatField())
+    max_bound = fof.ListField(fof.FloatField())
+    width = fof.IntField()
+    height = fof.IntField()
+
+
+def compute_orthographic_projection_images(
+    samples: foc.SampleCollection,
+    size: Vec2Tuple,
+    output_dir: str,
+    shading_mode: Literal["intensity", "rgb", "height", "none"] = "none",
+    rel_dir: str = None,
+    projection_normal: Vec3Tuple = None,
+    in_group_slice: str = None,
+    out_group_slice: str = None,
+    bounds: Tuple[Vec3Tuple, Vec3Tuple] = None,
+):
+    """Computes orthographic projection images for the point clouds in the given
+    collection.
+
+    This operation will add ``orthographic_projection_metadata``
+    field to each sample.
+
+    Examples::
+
+        import fiftyone as fo
+        import fiftyone.utils.utils3d as fou3d
+        import fiftyone.zoo as foz
+
+        dataset = foz.load_zoo_dataset("quickstart-groups")
+        view = dataset.select_group_slices("pcd")
+
+        fou3d.compute_orthographic_projection_images(
+            view,
+            (-1, 512),
+            "/tmp/proj"
+        )
+
+        session = fo.launch_app(dataset)
+
+    Args:
+        samples: a :class:`fiftyone.core.collections.SampleCollection`
+        size: the desired ``(width, height)`` for the generated maps. Either
+            dimension can be None or negative, in which case the appropriate
+            aspect-preserving value is used
+        output_dir: an output directory in which to store the images/maps
+        shading_mode: an optional field to specify how to shade the points.
+            "intensity" and "rgb" are only valid if the header contains
+            "rgb" flag, or else it'll default to "none" mode. In "none" mode,
+            all points are shaded white.
+        rel_dir (None): an optional relative directory to strip from each input
+            filepath to generate a unique identifier that is joined with
+            ``output_dir`` to generate an output path for each image. This
+            argument allows for populating nested subdirectories in
+            ``output_dir`` that match the shape of the input paths
+        projection_normal (None): (experimental) the normal vector of the plane
```

Comment:
I added the `projection_normal`. The use case is what if user's data isn't oriented z-up? They can provide `projection_normal=[0, 1, 0]` to project along y-axis instead.

The way it's implemented is to rotate `projection_normal -> [0, 0, 1]` so that the rest of the code can assume z-up.